### PR TITLE
Minor Eliza cleanup

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@andrewconner I think this preserves existing behavior, mind taking a quick look-over?

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala [hunk 1]
```diff
@@ -635,34 +635,21 @@ class MessagingCommander @Inject() (
...
+        log.warn(s"[validateRecipients] Could not determine what ${Json.stringify(unknown)} is supposed to be.")
+        None
     }
-
-    (rawUsers, rawNonUsers, rawOrgs)
+    (
+      rawCategorized.collect { case userId: ExternalId[User] @unchecked => userId },
+      rawCategorized.collect { case bc: BasicContact => bc },
+      rawCategorized.collect { case orgId: PublicId[Organization] @unchecked => orgId }
+    )
```
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala:652]: I'm not a huge fan of this `@unchecked`. I originally wrote it with lightweight wrapper classes that got around the type erasure issue, but it seemed heavy-handed.

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala [hunk 2]
```diff
@@ -635,34 +635,21 @@ class MessagingCommander @Inject() (
...
-                (acc._1, acc._2 :+ contact, acc._3)
-              }
-          }.getOrElse {
-            log.warn(s"[validateRecipients] Could not determine what ${recipient.toString} is supposed to be.")
-            (acc._1, acc._2, acc._3)
-          }
-      }
+    val rawCategorized = rawRecipients.flatMap {
+      case JsString(id) if id.startsWith("o") => Organization.validatePublicId(id).toOption
+      case JsString(id) if id.length == 36 => ExternalId.asOpt[User](id)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala:640]: A comment in the above code said that these `JsString` bits could be removed "after a good while". Any idea if we've reached that yet?

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala [hunk 3]
```diff
@@ -635,34 +635,21 @@ class MessagingCommander @Inject() (
   })
 
   def parseRecipients(rawRecipients: Seq[JsValue]): (Seq[ExternalId[User]], Seq[BasicContact], Seq[PublicId[Organization]]) = {
-    val (rawUsers, rawNonUsers, rawOrgs) = rawRecipients.foldRight((Seq.empty[ExternalId[User]], Seq.empty[BasicContact], Seq.empty[PublicId[Organization]])) { (recipient, acc) =>
-      recipient.asOpt[JsString] match { // heuristic to determine if it's a user or an org
-        case Some(JsString(id)) if id.startsWith("o") =>
-          (acc._1, acc._2, acc._3 :+ PublicId(id))
-        case Some(JsString(id)) if id.length == 36 =>
-          (acc._1 ++ ExternalId.asOpt[User](id), acc._2, acc._3)
-        case _ => // Starting in v XXXX `kind` is always sent (9/2/2015). Above can be removed after a good while.
```
Comment (andrewconner) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala:644]: To document our Slack convo, the ext isn't sending `kind` for users (and by extension, probably orgs), but _is_ for emails.

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala:652]: Aw. :(

**Comment by andrewconner** [comment]
Looks good to me, sans the string processing thing

**Pull request merged**